store/store.go

```diff
@@ -351,6 +351,26 @@ func (s *SnapUbuntuStoreRepository) getAllPurchases(channel string, auther Authe
 	return purchasesByID, nil
 }
 
+// getMustBuy determines if a snap requires a payment, based on if it is non-free and if the user has already bought it
+func getMustBuy(prices map[string]float64, purchases []*purchase) bool {
+	// if the snap is free, then it doesn't need purchasing
+	if len(prices) == 0 {
+		return false
+	}
+
+	// search through all the purchases for an app to see if there are any
+	// that are for the whole app, not in-app.
+	for _, purchase := range purchases {
+		// nil ItemSKU lets us know the purchase is for the whole app
+		if purchase.ItemSKU == "" {
```

Comment:
As a suggested guideline I tend to use myself, comments are often more clear if you make them inside the branch. For example, we could say something like this here:

`// Purchase is for the whole snap.`

Rather than repeating the conditional above in English in the comment.